### ISSUE TITLE
DataTable onSelect TypeScript types

### ISSUE THREAD

**Issue by oskari** [grommet/grommet#4784]
I think the types for onSelect -function are inferred wrong

### Expected Behavior

Using DataTable onSelect as seen on story [DataTable / Select](https://storybook.grommet.io/?path=/story/visualizations-datatable--select) should work.

### Actual Behavior

![image](https://user-images.githubusercontent.com/506596/100356564-12e20b00-2ffc-11eb-9bd9-8445f9fd7f4f.png)

### URL, screen shot, or Codepen exhibiting the issue

https://codesandbox.io/s/grommet-typescript-datatable-onselect-typing-hj11t?file=/src/App.tsx

### Your Environment

name | version
-|-
grommet |  2.16.0
react | 17.0.0
react-dom | 17.0.0
react-scripts | 3.3.0
styled-components | 5.2.1

- Grommet version: 2.16.0
- Browser Name and version: Firefox Developer Edition 83.0b9
- Operating System and version (desktop or mobile): macOS Catalina 10.15.7

**Comment by apawson** [comment]
Similarly, the new `Tip` component appears to be missing from `node_modules/grommet/index.d.ts`

![image](https://user-images.githubusercontent.com/31909523/100475060-0068df80-3097-11eb-9866-cddcb1024331.png)

**Comment by ShimiSun** [comment]
If you guys already have an idea of how to fix the interface file or the export, please feel free to make a contribution.

**Comment by apawson** [comment]
> If you guys already have an idea of how to fix the interface file or the export, please feel free to make a contribution.

https://github.com/grommet/grommet/pull/4785

**Comment by jcfilben** [comment]
[Here](https://github.com/grommet/grommet/pull/4483) is the initial PR where they type for `onSelect` in DataTable was defined. I tried to change the type from `onSelect?: (string | number)[];` to `onSelect?: (select: React.Dispatch<React.SetStateAction<(string | number)[]>>) => void;` which is the suggested type for the setter in useState but I was still getting an error when trying to use typescript in the DataTable / Select story. I have also found some information suggesting the use of Dispatch with useState typing but I haven't had success with that either. I am not sure what the correct type for `onSelect` is just wanted to give an update of the information I have found out about so far for this issue.
